src/openloco/windows/industry_list.cpp

```diff
@@ -1,16 +1,1272 @@
+                            }
+                            else
+                            {
+                                if (input::state() != input::input_state::scroll_left)
+                                {
+                                    self->min_width = window_size.width;
+                                    self->min_height = window_size.height;
+                                    self->max_width = window_size.width;
+                                    self->max_height = window_size.height;
+                                }
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    self->saved_view.mapX = 0;
+                    if (input::state() != input::input_state::scroll_left)
+                    {
+                        self->min_width = window_size.width;
+                        self->min_height = window_size.height;
+                        self->max_width = window_size.width;
+                        self->max_height = window_size.height;
+                    }
+                }
+            }
+            self->frame_no++;
+
+            self->call_prepare_draw();
+            WindowManager::invalidateWidget(WindowType::industryList, self->number, self->current_tab + common::widx::tab_industry_list);
+            if ((!input::has_flag(input::input_flags::tool_active)) || self->type != _toolWindowType || self->number != _toolWindowNumber)
+            {
+                WindowManager::close(self);
+            }
+        }
+
+        // 0x00458455
+        static void tooltip(FormatArguments& args, ui::window* self, widget_index widgetIndex)
+        {
+            args.push(string_ids::tooltip_scroll_new_industry_list);
+        }
+
+        // 0x004586EA
+        static void get_scroll_size(window* self, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight)
+        {
+            *scrollHeight = (4 + self->var_83C) / 5;
+            if (*scrollHeight == 0)
+                *scrollHeight += 1;
+            *scrollHeight *= 112;
+        }
+
+        // 0x00458C7F
+        static void drawIndustryThumb(gfx::drawpixelinfo_t* clipped, const openloco::industry_object* industryObj, int16_t x, int16_t y)
+        {
+            registers regs;
+            regs.cx = x;
+            regs.dx = y;
+            regs.edi = (uint32_t)clipped;
+            regs.ebp = (uint32_t)industryObj;
+            call(0x00458C7F, regs);
+        }
+
+        // 0x00458352
+        static void draw_scroll(ui::window* self, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex)
+        {
+            auto shade = colour::get_shade(self->colours[1], 4);
+            gfx::clear_single(*dpi, shade);
+
+            uint16_t xPos = 0;
+            uint16_t yPos = 0;
+            for (uint16_t i = 0; i < self->var_83C; i++)
+            {
+                word_E0C3C6 = 0xFFFF;
```

Comment:
This variable is only used by this function make it a local variable and remove the definition of the global.